packages/nitro-rpc-client/src/rpc-client.ts

```diff
@@ -57,17 +59,47 @@ export class NitroRpcClient implements RpcClientApi {
     return getAndValidateResult(res, "receive_voucher");
   }
 
-  public async WaitForObjective(objectiveId: string): Promise<void> {
-    return new Promise((resolve) => {
+  public async WaitForLedgerChannelToHaveStatus(
+    channelId: string,
+    status: ChannelStatus
+  ): Promise<void> {
+    const promise = new Promise<void>((resolve) => {
+      this.transport.Notifications.on(
+        "ledger_channel_updated",
+        (payload: LedgerChannelUpdatedNotification["params"]["payload"]) => {
+          if (payload.ID === channelId) {
+            this.GetLedgerChannel(channelId).then((l) => {
+              if (l.Status == status) resolve();
+            });
+          }
+        }
+      );
+    });
+    const ledger = await this.GetLedgerChannel(channelId);
+    if (ledger.Status == status) return;
+    return promise;
+  }
+
+  public async WaitForPaymentChannelToHaveStatus(
+    channelId: string,
+    status: ChannelStatus
+  ): Promise<void> {
+    const promise = new Promise<void>((resolve) => {
       this.transport.Notifications.on(
-        "objective_completed",
-        (params: ObjectiveCompleteNotification["params"]) => {
-          if (params["payload"] === objectiveId) {
-            resolve();
+        "payment_channel_updated",
+        (payload: PaymentChannelUpdatedNotification["params"]["payload"]) => {
+          if (payload.ID === channelId) {
+            this.GetLedgerChannel(channelId).then((l) => {
```

Comment:
Should this be `this.GetPaymentChannel` instead of `this.GetLedgerChannel`?

Reply:
Yes! good catch.